packages/scxa-marker-gene-heatmap/src/MarkerGeneHeatmap.js

```diff
@@ -96,19 +104,29 @@ const MarkerGeneHeatmap = (props) => {
       const color = `#000000`
       const zIndex = 5
       const splitCellTypeLabel = splitPhrase(heatmapOptionsProvider[heatmapType].labelsFormatter(cellTypeOrClusterId))
-      const shortenCellTypeLabel = splitCellTypeLabel.length <= 3 ? splitCellTypeLabel.join(`<br/>`) : splitCellTypeLabel.slice(0, 3).join(`<br/>`) + `...`
       plotLines.push({
         color,
         width: 2,
         value: plotLineAxisPosition,
         zIndex,
         label: heatmapType !== `multiexperimentcelltypes` && {
-          text: shortenCellTypeLabel,
+          text: splitCellTypeLabel,
           align: `right`,
           textAlign: `left`,
           rotation: heatmapType === `celltypes` ? -45 : 0,
           x: heatmapType === `celltypes` ? 3 : 0,
-          y: yOffset
+          y: yOffset,
+          formatter: () => {
+            const shortenCellTypeLabel = numberOfRows === 1 ?
+              splitCellTypeLabel.slice(0, 1).join(`<br/>`) + `...` :
```

Comment:
Is this code always put `...` at the end of a label when `numberOfRows === 1` even if the whole label text fits into the space?

Reply:
I added another condition if the splitCellTypeLabel just has one line then we don't add `...`.